src/components/Button/index.tsx

```diff
@@ -401,31 +400,28 @@ const FwdButton = ({
   size = 'large',
   tooltip,
   tooltipBaseId,
-  type: elementType = 'button',
+  type = 'button',
   variant = 'primary',
-  as: asProp,
   ...props
 }: ButtonProps) => {
   const as = useMemo(() => {
     if (disabled) return 'button'
-    if (asProp) return asProp
     if (href || download) return StyledLink
 
     return 'button'
-  }, [disabled, href, download, asProp])
+  }, [disabled, href, download])
 
   const displayProgressOnly = !children
 
   const iconMargin = extend || (progress && displayProgressOnly) ? 0 : 8
-  const type = as === 'button' ? elementType : undefined
 
   return (
     <Tooltip id={tooltipBaseId} text={tooltip}>
       <StyledButton
         {...props}
         href={href}
         download={download}
-        ref={innerRef}
+        ref={innerRef as Ref<HTMLButtonElement>}
```

Comment:
Instead maybe you can type innerRef as `HTMLButtonElement` avoiding this type assertion

Reply:
We could but with href prop it will be `HTMLAnchorElement`  and we will have more typing issue outside the component I guess ? So I can because i'm not a big fan of transform a `button` in a `a` but since the feature exist so far I think it's better to keep it the same way no ?

Reply:
okay I think we would change that when migrating the component fully